addons/arsenal/functions/fnc_fillRightPanel.sqf

```diff
@@ -42,101 +42,111 @@ if (!(ctrlShown _searchbarCtrl) || {ctrlFade _searchbarCtrl > 0}) then {
-            } foreach ([getArray (_weaponConfig >> _x >> "magazines"), getArray (_weaponConfig >> "magazines")] select (_x == "this"));
+// Get all weapons on unit
+private _weapons = [primaryWeapon GVAR(center), handgunWeapon GVAR(center), secondaryWeapon GVAR(center), binocular GVAR(center)];
+private _compatibleMagazines = [];
+private _muzzles = [];
 
-            // Magazine groups
-            {
-                private _magazineGroups = uiNamespace getVariable [QGVAR(magazineGroups), createHashMap];
-                private _magArray = _magazineGroups get (toLower _x);
-                {((_compatibleMagazines select _index) select _subIndex) pushBackUnique _x} forEach _magArray;
-            } foreach ([getArray (_weaponConfig >> _x >> "magazineWell"), getArray (_weaponConfig >> "magazineWell")] select (_x == "this"));
+// Get all compatibles magazines with unit's weapons
+{
 
+    _muzzles = _x call CBA_fnc_getMuzzles;
 
-        } foreach getArray (_weaponConfig >> "muzzles");
-    };
-} foreach [primaryWeapon GVAR(center), handgunWeapon GVAR(center), secondaryWeapon GVAR(center)];
-
-private _itemsToCheck = [];
-private _compatibleMagsPrimaryMuzzle = [];
-private _compatibleMagsSecondaryMuzzle = [];
+    // Get compatible magazines for primary & secondary muzzle (secondary muzzle is not guaranteed to exist)
+    // Assumption: One weapon can have two muzzles maximum
+    _compatibleMagazines pushBack [compatibleMagazines [_x, _muzzles param [0, ""]], compatibleMagazines [_x, _muzzles param [1, ""]]];
+} forEach _weapons;
 
+// This is used for the "compatible magazines" tab when a container is open
 private _allCompatibleMags = [];
+
 {
     _allCompatibleMags append (_x select 0);
     _allCompatibleMags append (_x select 1);
-} foreach _compatibleMagazines;
+} forEach _compatibleMagazines;
 
 private _ctrlPanel = _display displayCtrl IDC_rightTabContent;
+private _listnBox = _display displayCtrl IDC_rightTabContentListnBox;
+
+// Reset right panel content
+lbClear _ctrlPanel;
+lbClear _listnBox;
+
+_ctrlPanel lbSetCurSel -1;
+_listnBox lbSetCurSel -1;
+
+// Retrieve compatible items
+private _compatibleItems = [];
+private _itemsToCheck = [];
+private _compatibleMagsPrimaryMuzzle = [];
+private _compatibleMagsSecondaryMuzzle = [];
 
 switch (GVAR(currentLeftPanel)) do {
-    case IDC_buttonPrimaryWeapon : {
+    // If weapons or binoculars are chosen, get their compatible magazines & items
+    // Primary weapon
+    case IDC_buttonPrimaryWeapon: {
         _compatibleMagsPrimaryMuzzle = _compatibleMagazines select 0 select 0;
         _compatibleMagsSecondaryMuzzle = _compatibleMagazines select 0 select 1;
-        _compatibleItems = (primaryWeapon GVAR(center)) call bis_fnc_compatibleItems;
-        _itemsToCheck = GVAR(currentItems) select 18;
+        _compatibleItems = compatibleItems (_weapons select 0)/* call CBA_fnc_compatibleItems*/;
```

Comment:
Can be easily swapped out for `CBA_fnc_compatibleItems`.

Reply:
Decision has been taken to move forward with engine commands instead of CBA. If bugs are found with engine commands, ACE will revert back to the CBA functions.